.github/workflows/python-publish.yml

```diff
@@ -0,0 +1,39 @@
+# This workflow will upload a Python Package using Twine when a release is created
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#publishing-to-package-registries
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Upload Python Package
+
+on:
+  release:
+    types: [published]
+
+permissions:
+  contents: read
+
+jobs:
+  deploy:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python
+      uses: actions/setup-python@v3
+      with:
+        python-version: '3.10'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install build
+    - name: Build package
+      run: python -m build
+    - name: Publish package
+      uses: pypa/gh-action-pypi-publish@27b31702a0e7fc50959f5ad993c78deac1bdfc29
```

Comment:
I am wondering why it uses the [commit 27b31702a0e7fc50959f5ad993c78deac1bdfc29](https://github.com/pypa/gh-action-pypi-publish/commit/27b31702a0e7fc50959f5ad993c78deac1bdfc29) and not the releases on pypa/gh-action-pypi-publish e.g. [pypa/gh-action-pypi-publish@release/v1.5.1](https://github.com/pypa/gh-action-pypi-publish/releases/tag/v1.5.1) . The first is shown on [github doc](https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#publishing-to-package-registries) and the second is shown on [python doc](https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/#publishing-the-distribution-to-pypi-and-testpypi).

Reply:
I'm not sure! I just decided to use the one suggested by Github (listed [here](https://github.com/EcoExtreML/STEMMUS_SCOPE_Processing/actions/new)). I guess because the v1.5.1 release was not out yet, and the older one used Python 3.8.